database/ffldb/ff/rollback.go

```diff
@@ -33,91 +33,110 @@ func (s *flatFileStore) rollback(targetLocation *flatFileLocation) error {
 
 	// Grab the write cursor mutex since it is modified throughout this
 	// function.
-	cursor := s.writeCursor
-	cursor.Lock()
-	defer cursor.Unlock()
+	s.writeCursor.Lock()
+	defer s.writeCursor.Unlock()
 
 	// Nothing to do if the rollback point is the same as the current write
 	// cursor.
 	targetFileNumber := targetLocation.fileNumber
 	targetFileOffset := targetLocation.fileOffset
-	if cursor.currentFileNumber == targetFileNumber && cursor.currentOffset == targetFileOffset {
+	if s.writeCursor.currentFileNumber == targetFileNumber && s.writeCursor.currentOffset == targetFileOffset {
 		return nil
 	}
 
 	// If the rollback point is greater than the current write cursor then
 	// something has gone very wrong, e.g. database corruption.
-	if cursor.currentFileNumber < targetFileNumber ||
-		(cursor.currentFileNumber == targetFileNumber && cursor.currentOffset < targetFileOffset) {
+	if s.writeCursor.currentFileNumber < targetFileNumber ||
+		(s.writeCursor.currentFileNumber == targetFileNumber && s.writeCursor.currentOffset < targetFileOffset) {
 		return errors.Errorf("targetLocation is greater than the " +
 			"current write cursor")
 	}
 
 	// Regardless of any failures that happen below, reposition the write
 	// cursor to the target flat file and offset.
 	defer func() {
-		cursor.currentFileNumber = targetFileNumber
-		cursor.currentOffset = targetFileOffset
+		s.writeCursor.currentFileNumber = targetFileNumber
+		s.writeCursor.currentOffset = targetFileOffset
 	}()
 
 	log.Warnf("ROLLBACK: Rolling back to file %d, offset %d",
 		targetFileNumber, targetFileOffset)
 
-	// Close the current write file if it needs to be deleted. Then delete
-	// all files that are newer than the provided rollback file while
-	// also moving the write cursor file backwards accordingly.
-	if cursor.currentFileNumber > targetFileNumber {
-		cursor.currentFile.Lock()
-		if cursor.currentFile.file != nil {
-			_ = cursor.currentFile.file.Close()
-			cursor.currentFile.file = nil
+	// Close the current write file if it needs to be deleted.
+	if s.writeCursor.currentFileNumber > targetFileNumber {
+		s.writeCursor.currentFile.Lock()
+		if s.writeCursor.currentFile.file != nil {
+			_ = s.writeCursor.currentFile.file.Close()
```

Comment:
If we ignore the result anyway, the `_ =` is redundant (unless you explicitly want to show to the reader that you are ignoring the result - in that case also add a comment explaining why)